lib/settings-observer.js

```diff
@@ -91,7 +74,7 @@ class SettingsObserver {
    * @public
    */
   dispose() {
-    this._settings.removeListener('change', this._handleChange);
+    this._settings._watchHandlers = _.reject(this._settings._watchHandlers, n => n === this);
```

Comment:
this._settings._watchHandlers = []

Reply:
nope...this method just dispose/unwatch only 1 observer. i do not think we clear all our watcher list

Reply:
1 observer stores. 
  + keyPath
  + 1 handler
  + current value (we need this value to compare)
  ... others..

I applied this solution from the beginning.  E.g:
 watch('object.propA', handler1) => observer1
 watch('object.propA', handler2) => observer2
 watch('object.propB', handler3) => observer3
 watch('object.propB.propB1, handler4) => observer4
...

Reply:
Ah okay, this is removing from the list of handlers (which is a bit odd). The way this is accessed is a bit odd since _settings._watchHandlerrs are both suppose to be the private variable of the object, but they are not private of the observer, but private to the settings?

Reply:
"but they are not private of the observer, but private to the settings?" => yeah...exactly => the private property is confusing..

Reply:
In this case, the observer should call the settings or its container to remove it?

(ex: unwatch(observer)?

Reply:
actually i think we can remove observer.dispose method. Actually we do not want it i think... because in setting, we can setting.unwatch(observer).

I would like to keep observer.dispose because the testcases that relate to watch. In testcase, they call dispose after finished a case => Finally i think we should
 - remove observer dispose
 - update testcases (do not call observer.dispose(). call setting.unwatch instead)